pkg/controller/computenodeopenstack/computenodeopenstack_controller.go

```diff
@@ -554,6 +561,19 @@ func updateNodesStatus(c client.Client, kclient kubernetes.Interface, instance *
 			if previousNodeStatus == "None" {
 				break
 			}
+
+			// It might happen that the node switches to SchedulingDisabled while there
+			// is no machine deletion, e.g. when there is an issue with the machineconfig.
+			// Therefore check if the machine object has the DeletionTimestamp, or is
+			// already deleted, otherwise the node should not get drained
+			machineDeleted, err := computenodeopenstack.MachineFlaggedDeletedWithNodeName(c, kclient, instance, node.Name)
```

Comment:
perhaps the function name can be changed to something more descriptive, like: 
- hasDeletionTimestamp, or
- isMachineTagForRemoval, or
- isMachineBeingDeleted

or maybe simple "isNodeBeingRemove"

Reply:
changed to isMachineBeingDeleted